src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Formatting/FormattingTestBase.cs

```diff
@@ -183,16 +185,43 @@ private static (RazorCodeDocument, DocumentSnapshot) CreateCodeDocumentAndSnapsh
         {
             fileKind ??= FileKinds.Component;
             tagHelpers ??= Array.Empty<TagHelperDescriptor>();
+            if (fileKind == FileKinds.Component)
+            {
+                tagHelpers = tagHelpers.Concat(GetDefaultRuntimeComponents()).ToArray();
+            }
             var sourceDocument = text.GetRazorSourceDocument(path, path);
+
+            // Yes I know "BlazorServer_31 is weird, but thats what is in the taghelpers.json file
+            const string defaultImports = @"
+@using BlazorServer_31
+@using BlazorServer_31.Pages
+@using BlazorServer_31.Shared
+@using Microsoft.AspNetCore.Components
+@using Microsoft.AspNetCore.Components.Authorization
+@using Microsoft.AspNetCore.Components.Routing
+@using Microsoft.AspNetCore.Components.Web
+";
+
+            var importsPath = new Uri("file:///path/to/_Imports.razor").AbsolutePath;
+            var importsSourceText = SourceText.From(defaultImports);
+            var importsDocument = importsSourceText.GetRazorSourceDocument(importsPath, importsPath);
+            var importsSnapshot = new Mock<DocumentSnapshot>(MockBehavior.Strict);
+            importsSnapshot.Setup(d => d.GetTextAsync()).Returns(Task.FromResult(importsSourceText));
```

Comment:
Would `TestDocumentSnapshot.Create(importsPath, defaultImports)` work instead of mock?

Reply:
I tried to go down that path and remove all of the mocks, but it was a nightmare and broke a bunch of tests. I have the commits in another branch locally in case I feel like trying again :)

Reply:
> I tried to go down that path and remove all of the mocks, but it was a nightmare and broke a bunch of tests. I have the commits in another branch locally in case I feel like trying again :)

hell if you tried that's good enough for me! :shipit: !!!